src/Symfony/Component/Validator/ValidatorBuilder.php

```diff
@@ -285,42 +283,38 @@ public function setTranslationDomain($translationDomain)
     /**
      * {@inheritdoc}
      */
-    public function getValidator()
+    public function getLoaders()
     {
-        $metadataFactory = $this->metadataFactory;
-
-        if (!$metadataFactory) {
-            $loaders = array();
+        $loaders = array();
 
-            if (count($this->xmlMappings) > 1) {
-                $loaders[] = new XmlFilesLoader($this->xmlMappings);
-            } elseif (1 === count($this->xmlMappings)) {
-                $loaders[] = new XmlFileLoader($this->xmlMappings[0]);
-            }
+        foreach ($this->xmlMappings as $xmlMapping) {
+            $loaders[] = new XmlFileLoader($xmlMapping);
+        }
 
-            if (count($this->yamlMappings) > 1) {
-                $loaders[] = new YamlFilesLoader($this->yamlMappings);
-            } elseif (1 === count($this->yamlMappings)) {
-                $loaders[] = new YamlFileLoader($this->yamlMappings[0]);
-            }
+        foreach ($this->yamlMappings as $yamlMappings) {
+            $loaders[] = new YamlFileLoader($yamlMappings);
+        }
 
-            foreach ($this->methodMappings as $methodName) {
-                $loaders[] = new StaticMethodLoader($methodName);
-            }
+        foreach ($this->methodMappings as $methodName) {
+            $loaders[] = new StaticMethodLoader($methodName);
+        }
 
-            if ($this->annotationReader) {
-                $loaders[] = new AnnotationLoader($this->annotationReader);
-            }
+        if ($this->annotationReader) {
+            $loaders[] = new AnnotationLoader($this->annotationReader);
+        }
 
-            $loader = null;
+        return $loaders;
+    }
 
-            if (count($loaders) > 1) {
-                $loader = new LoaderChain($loaders);
-            } elseif (1 === count($loaders)) {
-                $loader = $loaders[0];
-            }
+    /**
+     * {@inheritdoc}
+     */
+    public function getValidator()
+    {
+        $metadataFactory = $this->metadataFactory;
 
-            $metadataFactory = new LazyLoadingMetadataFactory($loader, $this->metadataCache);
+        if (!$metadataFactory) {
+            $metadataFactory = new LazyLoadingMetadataFactory(new LoaderChain($this->getLoaders()), $this->metadataCache);
```

Comment:
$loaders = $this->getLoaders();
$metadataFactory = new LazyLoadingMetadataFactory(count($loaders) > 1 ? new LoaderChain($loaders) : $loaders[0], $this->metadataCache);